packages/sol/src/availabilitySlots.ts

```diff
@@ -0,0 +1,144 @@
+import { IAvailabilitySlot } from "@litespace/types";
+import { dayjs } from "@/dayjs";
+import { clone, flatten, min } from "lodash";
+
+/**
+  * Divide list of slots into sub-slots. It is used whenever a user wants to book a lesson
+  * e.g., slot from 8am-10pm, split by 30min => 8-8:30, 8:30-9, 9-9:30, 9:30-10.
+  * @param slots - list of the slots to be divided.
+  * @param duration - the duration of each subslot in minutes.
+  */
+export function getSubslotsBatch(
```

Comment:
Let's stick to `SubSlots` instead of `Subslots` for readability.  In the description we can write `sub-slot` instead of `subslot`. 

```suggestion
export function getSubSlotsBatch(
```